### DISCUSSION TITLE
Targeting a breakpoint range don't work as according to docs.

### DISCUSSION THREAD

**Discussion by J4v4Scr1pt** [tailwindlabs/tailwindcss#14591] [Help] [answered]
I'm trying to use the max [target breakpoints](https://tailwindcss.com/docs/responsive-design#targeting-a-breakpoint-range). But nothing is applied?

I have tried this: 
`md:max-xl:hidden`
`max-xl:hidden`
`max-[1280px]:hidden`

None of the above works, and I don't even get intellisense when hovering.

What is wrong?

Version of Tailwind: 3.4.13

**Comment by wongjn** [comment, marked as answer]
Consider checking:
- Tailwind is integrated into the project correctly.
- The `screens` config adheres to the restrictions for `max-*` variants to work: https://github.com/tailwindlabs/tailwindcss/pull/9558#restrictions
- Tailwind has compiled since you added the class.
- The `content` file globs sufficiently cover the file that the code appears in.
- [String concatenation/interpolation is not used to build the class name](https://tailwindcss.com/docs/content-configuration#dynamic-class-names). 

Otherwise, consider providing a project that reproduces the unexpected behavior.

**Reply by J4v4Scr1pt** [reply]
Thanks for the response! 🙏

I found the issue, but it is a bit strange IMO. Because all other things work as it should. 
We have a shareble tailwind config using [presets](https://tailwindcss.com/docs/presets).
I have tried moving the config from the npm package to the application, still same issue.
But if I remove the presents prop it works.

**EDIT:** I tried moving the exakt same config inside of the config replacing the preset. That also works. There seems to be something fishy about the preset prop.

project tailwind-config:

```
import { companyConfig } from '@company/components';
import type { Config } from 'tailwindcss';

export default {
	presets: [companyConfig ], // <--- works if I remove this.
	content: [
		'./components/**/*.{js,ts,jsx,tsx,mdx}',
		'./app/**/*.{js,ts,jsx,tsx,mdx}',
		'./node_modules/@company/components/**/*.{js,ts,jsx,tsx}',
		'./node_modules/@nextui-org/theme/dist/**/*.{js,ts,jsx,tsx}',
	],
} satisfies Config;
```


exported company-config from company npm package:

```
import {nextui} from '@nextui-org/react';

export default {
    theme: {
        fontFamily: {'sans': ["Poppins",'sans-serif']},
        colors: {
            transparent: 'transparent',
            current: 'currentColor',
            'temperature-red': '#ff5c61',
            'temperature-orange': '#ffa000',
            'temperature-yellow': '#ffd900',
            'temperature-lime': '#B3E818',
            'temperature-green': '#45CB1C',
        },
        extend: {
            gridTemplateColumns: {
                // Complex site-specific column configuration
                header: '72px calc(100vh - 72px) / max-content',
            },
        },
    },
    plugins: [
        nextui({
            addCommonColors: true,
            defaultTheme: 'light',
        }),
    ],
}
```

**Reply by J4v4Scr1pt** [reply]
I have created a [stackblitz](https://stackblitz.com/edit/vitejs-vite-vsecrp?file=src%2FApp.tsx,tailwind.config.js,companyTailwindConfig.ts&terminal=dev).
But this one do work with the presets.

For us, it don't work with the presets, but if I copy/paste the exact same code exported to be used in the tailwind config instead of the presets it works. Then there must be an issue with the exportet config from the library.

Then the question is, what in the world are wrong with our export of the config??! 🤔

**Reply by J4v4Scr1pt** [reply]
I have tried to export just an empty file from the lib (node_modules) with no tailwind configs at all. Still same issue.

I tried installing a external libs preset and it still works in the stackblitz.

**Reply by adamwathan** [reply]
Does your preset have a `screens` config that uses objects to configure complex custom screens?

**Reply by J4v4Scr1pt** [reply]
Yes we have, actually we are about to remove this, but we have not published the version without it yet.

It looks like this and is under the extend prop.
```
screens: {
                tall: {raw: '(min-height: 480px)'},
            },
```

**Reply by wongjn** [reply]
Yep, then `max-*` variants won't work. As per the second point [in my previous reply](https://github.com/tailwindlabs/tailwindcss/discussions/14591#discussioncomment-10841537):

> Consider checking:
> - […]
> - The `screens` config adheres to the restrictions for `max-*` variants to work: https://github.com/tailwindlabs/tailwindcss/pull/9558#restrictions

**Reply by J4v4Scr1pt** [reply]
Yes, I'm stupid 😅. We have removed it in the code, but the current version have it. That's why I thought we did not have it.

I'm not at the computer atm, but will try this asap. Thank you so much for all the help! 🙏

**Reply by J4v4Scr1pt** [reply]
Built a new version of our package without the screen prop and now all is ok!

Thx 4 all the help guiding my stupid head 🙏 !
